--- Makefile	2016-08-09 18:21:34.010621703 -0400
+++ Makefile.new	2016-08-09 18:22:07.267674403 -0400
@@ -281,10 +281,10 @@
 
 # O3 optimisation causes issues on ARM
 else ifeq ($(ARM), 1)
-   CFLAGS += -DNDEBUG $(ARCH) -O2 -fomit-frame-pointer -fstrict-aliasing
+   CFLAGS += -DNDEBUG $(arch) -O2 -fomit-frame-pointer -fstrict-aliasing
 
 else
-   CFLAGS += -DNDEBUG $(ARCH) -O3 -fomit-frame-pointer -fstrict-aliasing
+   CFLAGS += -DNDEBUG $(arch) -O3 -fomit-frame-pointer -fstrict-aliasing
 endif
 
 # extra options needed *only* for the osd files
