Description: uic: fix include path of QValidator-derived classes
Origin: upstream, http://code.qt.io/cgit/qt/qtbase.git/commit/?id=a4df9739a2898f49
Bug: https://bugs.debian.org/800375
Last-Update: 2015-11-23

--- a/src/tools/uic/qclass_lib_map.h
+++ b/src/tools/uic/qclass_lib_map.h
@@ -969,10 +969,10 @@
 QT_CLASS_LIB(QToolBar, QtWidgets, qtoolbar.h)
 QT_CLASS_LIB(QToolBox, QtWidgets, qtoolbox.h)
 QT_CLASS_LIB(QToolButton, QtWidgets, qtoolbutton.h)
-QT_CLASS_LIB(QValidator, QtWidgets, qvalidator.h)
-QT_CLASS_LIB(QIntValidator, QtWidgets, qvalidator.h)
-QT_CLASS_LIB(QDoubleValidator, QtWidgets, qvalidator.h)
-QT_CLASS_LIB(QRegExpValidator, QtWidgets, qvalidator.h)
+QT_CLASS_LIB(QValidator, QtGui, qvalidator.h)
+QT_CLASS_LIB(QIntValidator, QtGui, qvalidator.h)
+QT_CLASS_LIB(QDoubleValidator, QtGui, qvalidator.h)
+QT_CLASS_LIB(QRegExpValidator, QtGui, qvalidator.h)
 QT_CLASS_LIB(QScriptEngineDebugger, QtScriptTools, qscriptenginedebugger.h)
 QT_CLASS_LIB(QDesignerComponents, QtDesigner, qdesigner_components.h)
 QT_CLASS_LIB(QExtensionFactory, QtDesigner, default_extensionfactory.h)
