commit 991a05cc657345d24cb86658a2d3fbd25643bdb2
Author: Adam Jackson <ajax@redhat.com>
Date:   Wed Feb 10 15:11:54 2016 -0500

    layers: Install to CMAKE_INSTALL_LIBDIR

--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -75,7 +75,7 @@ else()
     target_link_Libraries(VkLayer_${target} layer_utils)
 	add_dependencies(VkLayer_${target} generate_vk_layer_helpers)
 	set_target_properties(VkLayer_${target} PROPERTIES LINK_FLAGS "-Wl,-Bsymbolic")
-	install(TARGETS VkLayer_${target} DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+	install(TARGETS VkLayer_${target} DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endmacro()
 endif()
 
@@ -143,7 +143,7 @@ if (WIN32)
     set_target_properties(layer_utils_static PROPERTIES OUTPUT_NAME layer_utils)
     target_link_libraries(layer_utils)
 else()
-    install(TARGETS layer_utils DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+    install(TARGETS layer_utils DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 add_vk_layer(core_validation core_validation.cpp vk_layer_table.cpp)
 add_vk_layer(device_limits device_limits.cpp vk_layer_table.cpp vk_layer_utils.cpp)
