commit d9add88314908456d458da18f48d0a3a2c73789e
Author: Adam Jackson <ajax@redhat.com>
Date:   Wed Feb 10 14:52:34 2016 -0500

    layers: Don't set an rpath

--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -31,9 +31,6 @@ set(LAYER_JSON_FILES
     VkLayer_device_limits
     )
 
-set(VK_LAYER_RPATH /usr/lib/x86_64-linux-gnu/vulkan/layer:/usr/lib/i386-linux-gnu/vulkan/layer)
-set(CMAKE_INSTALL_RPATH ${VK_LAYER_RPATH})
-
 if (NOT WIN32)
     # extra setup for out-of-tree builds
     if (NOT (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR))
