Description: Make the build reproducible
Author: Stephen Kitt <skitt@debian.org>

--- a/Quake/host.c
+++ b/Quake/host.c
@@ -221,7 +221,6 @@
 {
 	Con_Printf ("Quake Version %1.2f\n", VERSION);
 	Con_Printf ("QuakeSpasm Version %1.2f.%d\n", QUAKESPASM_VERSION, QUAKESPASM_VER_PATCH);
-	Con_Printf ("Exe: "__TIME__" "__DATE__"\n");
 }
 
 /* cvar callback functions : */
@@ -844,7 +843,6 @@
 	NET_Init ();
 	SV_Init ();
 
-	Con_Printf ("Exe: "__TIME__" "__DATE__"\n");
 	Con_Printf ("%4.1f megabyte heap\n", host_parms->memsize/ (1024*1024.0));
 
 	if (cls.state != ca_dedicated)
