Description: fix pthread linking
Origin: upstream, https://code.qt.io/cgit/qt/qtwebkit.git/commit/?id=5f359baacdf92fab
Last-Update: 2016-03-06

--- a/Tools/qmake/mkspecs/features/default_post.prf
+++ b/Tools/qmake/mkspecs/features/default_post.prf
@@ -201,6 +201,7 @@
         linkAgainstLibrary($$library, $$eval(WEBKIT.$${library_identifier}.root_source_dir))
         LIBS += $$eval(WEBKIT.$${library_identifier}.dependent_libs)
     }
+    posix:!darwin: LIBS += -lpthread
 }
 
 creating_module {
