--- a/engine/Makefile
+++ b/engine/Makefile
@@ -2077,15 +2077,16 @@
 
 utils: httpserver iqm
 
-prefix ?= /usr/local
+prefix ?= /usr
 exec_prefix ?= $(prefix)
-bindir ?= $(exec_prefix)/bin
+bindir ?= $(exec_prefix)/games
 sbindir ?= $(exec_prefix)/sbin
 INSTALL ?= install
 INSTALL_PROGRAM ?= $(INSTALL)
 INSTALL_DATA ?= ${INSTALL} -m 644
-install: sv-rel gl-rel mingl-rel qcc-rel
+install: sv-rel gl-rel mingl-rel qcc-rel vk-rel
 	$(INSTALL_PROGRAM) $(RELEASE_DIR)/$(EXE_NAME)-gl $(DESTDIR)$(bindir)/$(EXE_NAME)-gl
 	$(INSTALL_PROGRAM) $(RELEASE_DIR)/$(EXE_NAME)-mingl $(DESTDIR)$(bindir)/$(EXE_NAME)-mingl
 	$(INSTALL_PROGRAM) $(RELEASE_DIR)/$(EXE_NAME)-sv $(DESTDIR)$(bindir)/$(EXE_NAME)-sv
+	$(INSTALL_PROGRAM) $(RELEASE_DIR)/$(EXE_NAME)-vk $(DESTDIR)$(bindir)/$(EXE_NAME)-vk
 	$(INSTALL_PROGRAM) $(RELEASE_DIR)/fteqcc $(DESTDIR)$(bindir)/fteqcc
