Description: Provide repo_autopattern symbol for non-SUSE builds
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Forwarded: https://github.com/openSUSE/libsolv/issues/88

--- a/ext/CMakeLists.txt
+++ b/ext/CMakeLists.txt
@@ -53,12 +53,12 @@
 ENDIF (ENABLE_SUSEREPO OR ENABLE_RPMMD OR ENABLE_RPMDB)
 ENDIF (ENABLE_COMPLEX_DEPS)
 
-IF (SUSE)
+IF (ENABLE_RPMMD)
     SET (libsolvext_SRCS ${libsolvext_SRCS}
 	repo_autopattern.c)
     SET (libsolvext_HEADERS ${libsolvext_HEADERS}
 	repo_autopattern.h)
-ENDIF (SUSE)
+ENDIF (ENABLE_RPMMD)
 
 IF (ENABLE_COMPS)
     SET (libsolvext_SRCS ${libsolvext_SRCS}
