Index: llvm-toolchain-snapshot_3.8~svn255217/lldb/third_party/Python/module/unittest2/unittest2/runner.py
===================================================================
--- llvm-toolchain-snapshot_3.8~svn255217.orig/lldb/third_party/Python/module/unittest2/unittest2/runner.py
+++ llvm-toolchain-snapshot_3.8~svn255217/lldb/third_party/Python/module/unittest2/unittest2/runner.py
@@ -171,9 +171,9 @@ class TextTestRunner(unittest.TextTestRu
         if hasattr(result, 'separator2'):
             self.stream.writeln(result.separator2)
         run = result.testsRun
-        self.stream.writeln("Ran %d test%s in %.3fs" %
-                            (run, run != 1 and "s" or "", timeTaken))
-        self.stream.writeln()
+#        self.stream.writeln("Ran %d test%s in %.3fs" %
+#                            (run, run != 1 and "s" or "", timeTaken))
+#        self.stream.writeln()
         
         expectedFails = unexpectedSuccesses = skipped = passed = failed = errored = 0
         try:
