Index: llvm-toolchain-snapshot_3.6~svn217023/utils/TableGen/CodeEmitterGen.cpp
===================================================================
--- llvm-toolchain-snapshot_3.6~svn217023.orig/utils/TableGen/CodeEmitterGen.cpp
+++ llvm-toolchain-snapshot_3.6~svn217023/utils/TableGen/CodeEmitterGen.cpp
@@ -230,6 +230,9 @@ void CodeEmitterGen::run(raw_ostream &o)
   const std::vector<const CodeGenInstruction*> &NumberedInstructions =
     Target.getInstructionsByEnumValue();
 
+  o << "// Undef for HURD\n";
+  o << "#ifdef EIEIO\n#undef EIEIO\n#endif\n";
+
   // Emit function declaration
   o << "uint64_t " << Target.getName();
   o << "MCCodeEmitter::getBinaryCodeForInstr(const MCInst &MI,\n"
